paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/CheckCloneResultAndCopySnapshotOperator.java

```diff
@@ -0,0 +1,192 @@
+
+    private static final Logger LOG =
+            LoggerFactory.getLogger(CheckCloneResultAndCopySnapshotOperator.class);
+
+    private final FileIO sourceTableFileIO;
+    private final FileIO targetTableFileIO;
+    private final SnapshotManager targetTableSnapshotManager;
+    private final boolean copyInDifferentCluster;
+    private final Path sourceTableRootPath;
+    private final Path targetTableRootPath;
+    private final List<CloneFileInfo> snapshotOrTagFiles;
+    private final CloneType cloneType;
+    private long checkedFileCount;
+
+    public CheckCloneResultAndCopySnapshotOperator(
+            FileStoreTable sourceTable,
+            FileStoreTable targetTable,
+            boolean copyInDifferentCluster,
+            Path sourceTableRootPath,
+            Path targetTableRootPath,
+            CloneType cloneType) {
+        super();
+        this.sourceTableFileIO = sourceTable.fileIO();
+        this.targetTableFileIO = targetTable.fileIO();
+        this.targetTableSnapshotManager = targetTable.snapshotManager();
+        this.copyInDifferentCluster = copyInDifferentCluster;
+        this.sourceTableRootPath = sourceTableRootPath;
+        this.targetTableRootPath = targetTableRootPath;
+        this.snapshotOrTagFiles = new ArrayList<>();
+        this.cloneType = cloneType;
+        this.checkedFileCount = 0;
+    }
+
+    @Override
+    public void setup(
+            StreamTask<?, ?> containingTask,
+            StreamConfig config,
+            Output<StreamRecord<CloneFileInfo>> output) {
+        super.setup(containingTask, config, output);
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+    }
+
+    @Override
+    public void processElement(StreamRecord<CloneFileInfo> streamRecord) throws Exception {
+        output.collect(streamRecord);
+
+        CloneFileInfo cloneFileInfo = streamRecord.getValue();
+        if (cloneFileInfo.isSnapshotOrTagFile()) {
+            snapshotOrTagFiles.add(cloneFileInfo);
+            return;
+        }
+
+        Path targetFilePath =
+                new Path(
+                        targetTableRootPath.toString()
+                                + cloneFileInfo.getFilePathExcludeTableRoot());
+
+        LOG.info(
+                "Begin check the {} file's size, targetFilePath is {}.",
+                ++checkedFileCount,
+                targetFilePath);
+
+        //        checkState(
+        //                cloneFileInfo.getFileSize() ==
+        // targetTableFileIO.getFileSize(targetFilePath),
+        //                "Copy file error. clone file origin size is : "
+        //                        + cloneFileInfo.getFileSize()
+        //                        + ", target file path is : "
+        //                        + targetFilePath
+        //                        + ", and target file size is  : "
+        //                        + targetTableFileIO.getFileSize(targetFilePath));
```

Comment:
What are you checking? I don't see any code for checking.

Reply:
done, check file size